tests/components/logentries/test_init.py

```diff
@@ -1,82 +1,74 @@
-        mock_requests.exceptions.RequestException = self.mock_request_exception
-        config = {"logentries": {"token": "token"}}
-        self.hass.bus.listen = mock.MagicMock()
-        setup_component(self.hass, logentries.DOMAIN, config)
-        self.handler_method = self.hass.bus.listen.call_args_list[0][0][1]
 
-    @mock.patch.object(logentries, "requests")
-    @mock.patch("json.dumps")
-    def test_event_listener(self, mock_dump, mock_requests):
-        """Test event listener."""
-        mock_dump.side_effect = lambda x: x
-        self._setup(mock_requests)
+@pytest.fixture
+def mock_requests():
+    """Mock requests."""
+    with patch.object(logentries, "requests") as mock_requests:
+        yield mock_requests
 
-        valid = {"1": 1, "1.0": 1.0, STATE_ON: 1, STATE_OFF: 0, "foo": "foo"}
-        for in_, out in valid.items():
-            state = mock.MagicMock(
-                state=in_, domain="fake", object_id="entity", attributes={}
-            )
-            event = mock.MagicMock(data={"new_state": state}, time_fired=12345)
-            body = [
-                {
-                    "domain": "fake",
-                    "entity_id": "entity",
-                    "attributes": {},
-                    "time": "12345",
-                    "value": out,
-                }
-            ]
-            payload = {
-                "host": "https://webhook.logentries.com/noformat/logs/token",
-                "event": body,
+
+async def test_event_listener(hass, mock_dump, mock_requests):
+    """Test event listener."""
+    mock_dump.side_effect = lambda x: x
+    mock_post = mock_requests.post
+    mock_requests.exceptions.RequestException = Exception
+    config = {"logentries": {"token": "token"}}
+    hass.bus.listen = MagicMock()
+    assert await async_setup_component(hass, logentries.DOMAIN, config)
+    handler_method = hass.bus.listen.call_args_list[0][0][1]
+
+    valid = {"1": 1, "1.0": 1.0, STATE_ON: 1, STATE_OFF: 0, "foo": "foo"}
+    for in_, out in valid.items():
+        state = MagicMock(state=in_, domain="fake", object_id="entity", attributes={})
+        event = MagicMock(data={"new_state": state}, time_fired=12345)
+        body = [
+            {
+                "domain": "fake",
+                "entity_id": "entity",
+                "attributes": {},
+                "time": "12345",
+                "value": out,
             }
-            self.handler_method(event)
-            assert self.mock_post.call_count == 1
-            assert self.mock_post.call_args == mock.call(
-                payload["host"], data=payload, timeout=10
-            )
-            self.mock_post.reset_mock()
+        ]
+        payload = {
+            "host": "https://webhook.logentries.com/noformat/logs/token",
+            "event": body,
+        }
+        handler_method(event)
+        assert mock_post.call_count == 1
+        assert mock_post.call_args == call(payload["host"], data=payload, timeout=10)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)